### PR TITLE
bump containernetworking/cni to v0.8.0

### PR DESCRIPTION
**Pull request by aojea**
last tagged release of this module was 0.7.1, however,
there are more than 100 commits between master and
that tag:

~/go/src/github.com/containernetworking/cni$ git log --pretty=oneline
master...v0.7.1 | wc
    104     760    9234

Since this project does not seems to follow a release cadence, we
should keep tracking the mater branch to benefit from the last
fixes and improvements.

Signed-off-by: Antonio Ojea <antonio.ojea.garcia@gmail.com>

### CODE REVIEW

**Comment by aojea** [comment]
the diff show some bugfixes and improvements, that may fix some of the issues I can observe in the K8S CI from time to time related to the CNI

/cc @liggit @dims @bentheelder

**Comment by mikebrow** [comment]
We generally move in cadence with the rest of the stack that is dependent on CNI, which has not happened since around august of last year, and of course not just CNI but also the plugins.

We, containerd are in beta on v1.4 and kubernetes is in beta on v1.19.  Both will soon move on to a new cycle which would be a great time to bump CNI. Sort of last min if meant for 1.4.0...

See:
- https://github.com/kubernetes/kubernetes/blob/master/go.mod#L213
- https://github.com/containerd/cri/blob/master/vendor.conf#L91-L92

Is kubernetes moving up? before they release?

Should go over the changes to see what changes are necessary in this repo, cri, etc.. to use/take advantage and see if there are expected regressions that we may not be testing yet. 

The new cached config apis look interesting.

**Comment by aojea** [comment]
> Is kubernetes moving up? before they release?

Kubernetes no longer depends on CNI directly since the CRI interface was implemented, 
> https://github.com/kubernetes/kubernetes/blob/master/go.mod#L213

if you do a grep in the k/k repo you can see that the CNI code is only being used by the `dockershim`

```
grep -r containernetworking | grep -v -E "vendor|CHANGELOG" | grep -v dockershim
go.mod:	github.com/containernetworking/cni v0.7.1
go.mod:	github.com/containernetworking/cni => github.com/containernetworking/cni v0.7.1
Binary file .git/index matches
go.sum:github.com/containernetworking/cni v0.7.1 h1:fE3r16wpSEyaqY4Z4oFrLMmIGfBYIKpPrHK31EJ9FzE=
go.sum:github.com/containernetworking/cni v0.7.1/go.mod h1:LGwApLUm2FpoOfxTDEeq8T9ipbpZ61X79hmU3w8FmsY=
```

... and you can see that CRI-O is using the latest versions

https://github.com/cri-o/cri-o/blob/4c4a2502f53842edb5b680c4cd55256871cdf365/go.mod#L11-L12

> Should go over the changes to see what changes are necessary in this repo, cri, etc.. to use/take advantage and see if there are expected regressions that we may not be testing yet.

I recognize I didn't do an exhaustive look at the diff between master and the last tagged version, but you can observe that there several bugfixes and improvements, like the cache one that you mention. 

The CNI code is not huge,that's why I thought it may be updated directly instead of cherrypicking the changes

**Comment by aojea** [comment]
@mikebrow another important thing is that there are also CVE affection previous cni versions
https://github.com/kubernetes/kubernetes/issues/91507

```
Many container networking implementations are affected, including:

CNI Plugins maintained by the containernetworking team, prior to version 0.8.6 (CVE-2020-10749) (See containernetworking/plugins#484)
```

**Comment by aojea** [comment]
/hold

let's see if the cni project can tag a new version
xref https://github.com/kubernetes/kubernetes/pull/91896#issuecomment-640513584

**Comment by mikebrow** [comment]
tag would be nice.. and a tested / recommended plugin release too.  

I did check cri-o before my post yesterday.. I saw they were on a different commit..2020 03 04 than this proposed commit 2020 05 13.  

I think docker shim is still used and expected to work. But that is just one of.. Here's a grep over the kubernetes eco system against github.com/containernetworking/cni
https://cs.k8s.io/?q=github.com%2Fcontainernetworking%2Fcni&i=nope&files=&repos=

I support moving up ... and I'm concerned about timing, regressions, potential cves, .. tough spot. sig-node is understaffed and it's close to the end of beta time...

File: cni_test.go [hunk 1]
```diff
@@ -223,6 +223,10 @@ func (m *MockCNI) CheckNetwork(ctx context.Context, net *cnilibrary.NetworkConfi
 	args := m.Called(net, rt)
 	return args.Error(0)
 }
+func (m *MockCNI) GetNetworkCachedConfig(net *cnilibrary.NetworkConfig, rt *cnilibrary.RuntimeConf) ([]byte, *cnilibrary.RuntimeConf, error) {
```
Comment (mikebrow) [inline cni_test.go:226]: blank line nit

Comment (aojea) [reply cni_test.go:227]: upps, :sweat_smile: 
done

**Comment by mikebrow** [comment]
When we merge this let's tag it..

**Comment by AkihiroSuda** [comment]
> When we merge this let's tag it..

Tagged as v1.0.1 https://github.com/containerd/go-cni/releases/tag/v1.0.1

cc @dmcgowan

**Pull request merged**